atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,264 @@
+"""
+Copyright (c) 2015-2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+
+Pre build plugin which injects custom yum repositories in dockerfile.
+"""
+import os
+import shutil
+from io import StringIO
+from urllib.parse import urlparse
+
+from atomic_reactor.config import get_koji_session
+from atomic_reactor.constants import YUM_REPOS_DIR, RELATIVE_REPOS_PATH, INSPECT_CONFIG, \
+    PLUGIN_RESOLVE_COMPOSES_KEY
+from atomic_reactor.dirs import BuildDir
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.util import is_scratch_build, map_to_user_params, \
+    allow_repo_dir_in_dockerignore, render_yum_repo
+from atomic_reactor.utils.yum import YumRepo
+
+
+class InjectYumReposPlugin(PreBuildPlugin):
+    key = "inject_yum_repos"
+    is_allowed_to_fail = False
+
+    args_from_user_params = map_to_user_params(
+        "target:koji_target",
+        "repourls:yum_repourls",
+    )
+
+    def validate_yum_repo_files_url(self):
+        if not self.allowed_domains:
+            return
+        errors = []
+
+        for repourl in self.repourls:
+            repo_domain = urlparse(repourl).netloc
+            if repo_domain not in self.allowed_domains:
+                errors.append('Yum repo URL {} is not in list of allowed domains: {}'
+                              .format(repourl, self.allowed_domains))
+
+        if errors:
+            raise ValueError('Errors found while checking yum repo urls: \n{}'
+                             .format('\n'.join(errors)))
+
+    def _final_user_line(self):
+        user = self._find_final_user()
+        if user:
+            return user
+
+        if not self.workflow.dockerfile_images.base_from_scratch:
+            # OSBS2 TBD: decide if we need to inspect a specific platform
```

Comment:
not sure what advantage inspecting for specific architecture will offer here since we're already looking up `USER` in platform-specific `Dockerfile` in `self._find_final_user()`

Reply:
I've reached to the following conclusion after observing another PR. We can inspect any platform since the `User` should be same for all platforms.